### ISSUE TITLE
[Dungeon] Zul Farrak - Sometimes Seargent Bly gets reset after first wave/or doesnt leave the cage at all [$25]

### ISSUE THREAD

**Issue by 55Honey** [azerothcore/azerothcore-wotlk#6696]
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [x] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Sergeant Bly  (https://wotlk.evowow.com/?npc=7604) is sometimes refusing to leave the cage, or "gets reset" qand runs back tot the cage after 1st wave of mobs dies.
Seems something is breaking the script, doesnt happen every time tho, sometimes it works fine.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Seargent Bly should leave the cage when cages are unlocked and stay out of the cage so players can interact with him to progress the dungeon.

##### STEPS TO REPRODUCE THE PROBLEM:

1. Do Zul Farrak dungeon up to the Cages part.
2. Unlock the cages and see if the script works correctly

##### EXTRA NOTES:

https://github.com/chromiecraft/chromiecraft/issues/1066

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/cecbd604d3f57312165cc53f8ed40feae51ee45f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

**Comment by 55Honey** [comment]
[gurveleg:](https://github.com/gurveleg:)
> It seem that engaging the mobs too fast, before they come up the stairs and aggro on npcs is one cause of them bugging.
We did 3 runs when we waited for mobs to come up the stairs so the NPCs aggroed on them, and it didnt break the script.

**Comment by da-stealth** [comment]
> [gurveleg:](https://github.com/gurveleg:)
> 
> > It seem that engaging the mobs too fast, before they come up the stairs and aggro on npcs is one cause of them bugging.
> > We did 3 runs when we waited for mobs to come up the stairs so the NPCs aggroed on them, and it didnt break the script.

I concur. If you start attacking the mobs before Weegil Blastfuse (the goblin) says "Oh no! Here they come!", they seem to get stuck. It seems like the flag condition doesn't get met or trigger unless he says this phrase.

**Comment by UltraNix** [comment]
I don't know what to do to reproduce it :/

**Comment by 55Honey** [comment]
Engage mobs manually while they're running up the stairs, before the alles NPCs can reach them.

In general when being slow the script works. Acting super fast breaks it.

**Comment by UltraNix** [comment]
I was trying that - no result.

**Comment by syssneck** [comment]
I tried two times to bug it out. I fought only downstairs, killed the mobs really fast, but the event worked each time and the goblin opened the door. On the second try I thought it bugged, when I was talking to him upstairs, but after waiting for half a minute or so, his usual message: "Let's move forward" came, and they all went downstairs. I have no clue on how to reproduce this issue.

**Comment by temperrr** [comment]
I'll come back to this tomorrow morning and see if I can break it consistently

**Comment by temperrr** [comment]
UNSUCCESFUL test 1: aggressively pressing the option to "settle things on the battlefield" and actively killings monsters before they reach them.  I t goes till "lets move forward" they move forward I kill boss. No more option to talk to Sergeant Bly. They are stuck standing on a line at bottom of the staircase. ( i waited 9 minutes.) This option should not be available before you kill the boss.

UNSUCCESFUL Test 2:  Agressively pressing the option to "settle things on the battlefield" and taking it slow (e.g. letting monsters come up the stairs towards the npcs etc). All went great and pulled boss with them. Waited 11 minutes. They did not proceed to bonfire but standed at bottom of stairs not being able to interract with.

Kinda Succesful.... (still wrong)  Test 3: Not aggressive pressing the option to "settle things on the battlefield" and instead just taking it slow. They proceeded towards the bonfire this time. I got two options to "settle things on the battlefield" Neither of them engaged the fight with them. I could however talk to goblin to open the door. and that did happen. However, when he gets in combat, he will not open the door, unable to talk to him again and he will stand by the bonfire.

UnsuccesfulTest 4:  Not aggressive pressing the option to "settle things on the battlefield" and instead rushing to kill everything.  Waited 5 minutes. Stuck at bottom of stairs. No option to talk and they did not proceed to bonfire.

![image](https://user-images.githubusercontent.com/20357406/132814706-5aa02c0d-c821-434f-a9d1-46ba3855de9e.png)


Things that are not working correctly:
Sergeant bly constantly casts some kind of beserker stance when engaging enemies. sometimes he is stuck in that animation.
the option "to settle things on the battlefield" should only appear when you killed boss. (sometimes the option appears twice after fighting boss...)
The goblin will not open door when he gets in combat in the way to the door and bugs out.

Lets just say I have trouble to get it to work at all, so difficult to see where it goes wrong. But hopefully my notes are something that can be look at maybe thats the cause of events breaking?

**Comment by Efymer** [comment]
I've tried multiple times to reproduce the issue using the methods mentioned here (engaging super fast, not letting the NPCs get in combat, etc, etc.) and it worked perfectly fine each time 😕 

Throwing a thought here: since the script plays a lot with factions, could it be an issue of the cross faction module or it has been reproduced in a clean AC branch?

**Comment by syssneck** [comment]
Well @temperrr also tested it on his local AC, but compared to us, he was able to reproduce it. On what branch did you guys test it? I was on 1510752f64b48 with no success of reproducing the issue. I may give it another shot later.

**Comment by temperrr** [comment]
Clean ac aca8e5697177 2021-09-03

**Comment by syssneck** [comment]
Okay, It also broke for me now. They are not moving and there is no speech bubble on any of them. They went downstairs, fought one enemy and as the enemy died, they went back to their position and didn't move since:
![grafik](https://user-images.githubusercontent.com/68868567/132857080-78bfe90b-ce3b-44cc-8b3f-3ccfaad00a97.png)

There was already a big delay between wave 1 and 2 (I spoke to Bly again after a minute or so) and after wave 2 (also a minute and I spammed that gossip option upstairs). I killed both waves pretty fast and engaged downstairs before Sergeant Bly and his crew did.
